File: lab/c/matematica_e_operadores/mmc.c
#include <stdio.h>
#include <stdlib.h>

int mdc(int a, int b) {
    a = abs(a);
    b = abs(b);

    if (a == 0 && b == 0) {
        printf("Erro: MDC indefinido para (0,0)\n");
        exit(1);
    }

    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int mmc(int a, int b) {
    if (a == 0 || b == 0) {
        return 0;
    }
    return abs(a * b) / mdc(a, b);
}

int main() {
    printf("%d\n", mmc(12, 18)); // 36
    return 0;
}

File: lab/javascript/matematica_e_operadores/mmc.js
function mdc(a, b) {
    a = Math.abs(a);
    b = Math.abs(b);

    if (a === 0 && b === 0) {
        throw new Error("MDC indefinido para (0,0)");
    }

    while (b !== 0) {
        let temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

function mmc(a, b) {
    if (a === 0 || b === 0) {
        return 0;
    }
    return Math.abs(a * b) / mdc(a, b);
}

console.log(mmc(12, 18)); // 36
